CMSgov/onemac
# Reset Latest RAI Timestamp with specific actions and statuses

Story: https://qmacbis.atlassian.net/browse/OY2-24635

### Details

This PR adds logic to reset the `latestRaiResponseTimestamp` value of a package when **EITHER** a package is put into "Pending RAI" (CMS) or "RAI Issued" (State) **OR** the RAI component for a package has a status other than "Submitted". 

This should cover two scenarios:
1. The status of an RAI Response changes due to OneMAC package actions like enabling withdraw, requesting withdraw, or the withdraw itself.
2. The status of a OneMAC package changes back into a Pending RAI / RAI Issued status due to SEA Tool intervention.

### Changes

- Addition of condition that a package must NOT be in the Pending RAI / RAI Issued status to set the `latestRaiResponseTimestamp` to the timestamp of the most recent RAI Response for a package
- Addition of an else block to reset `latestRaiResponseTimestamp` to `emptyField`

### Implementation Notes

- Added one-seed data to show scenario 2 (package actions resetting field)

### Test Plan

> ℹ️ - This is testable locally, but I'm not able to dissect why one-seed is populating my local with the new test data but not the branch's test environment.

1. Spin up the API, one-stream, and front-end locally
2. Log in as a CMS user
3. Find MD-23-3333-VM. This package has been set up with an RAI Response and has the Enable Withdraw package action available.
4. Observe the timestamp present in the details section
5. Use the package action to enable withdraw of rai response
6. Observe that the date has been reset and appears as "-- --" in the details section
